test/libsyncengine/reconciliation/platform_inconsistency_checker/testplatforminconsistencycheckerworker.cpp

```diff
@@ -276,4 +276,85 @@ void TestPlatformInconsistencyCheckerWorker::testExecute() {
 #endif
 }
 
+void TestPlatformInconsistencyCheckerWorker::testNameSizeLocalTree() {
+    initUpdateTree(ReplicaSide::Local);
+   
+    _syncPal->_platformInconsistencyCheckerWorker->execute();
+
+    CPPUNIT_ASSERT(_syncPal->updateTree(ReplicaSide::Local)->exists("testNode"));
+    CPPUNIT_ASSERT(_syncPal->updateTree(ReplicaSide::Local)->exists("aNode"));
+    CPPUNIT_ASSERT(_syncPal->updateTree(ReplicaSide::Local)->exists("ANode"));
+    CPPUNIT_ASSERT(!_syncPal->updateTree(ReplicaSide::Local)->exists("longNameANode"));
+    CPPUNIT_ASSERT(!_syncPal->updateTree(ReplicaSide::Local)->exists("testNode2"));
+    CPPUNIT_ASSERT(!_syncPal->updateTree(ReplicaSide::Local)->exists("bNode"));
+    CPPUNIT_ASSERT(!_syncPal->updateTree(ReplicaSide::Local)->exists("BNode"));
+}
+
+void TestPlatformInconsistencyCheckerWorker::initUpdateTree(ReplicaSide side) {
+    /* Initial tree structure:
+     *  |
+     *  +-- /test (dir) CREATE
+     *  |   |
+     *  |   +-- a.txt (file) CREATE
+     *  |   +-- A.txt (file) CREATE
+     *  |   +-- aaaaaaaaaaaaaaaaaaaa...aaaaaaaaaaaaaaaaaa.txt (file)  [maxNameLengh +1] CREATE
+     *  |
+     *  +-- /testDiraaaaaaaaaaaaaaa...aaaaaaaaaaaaaaaaa  (dir)  [maxNameLengh +1] MOVE
+     *  |   |
+     *  |   +-- b.txt (file) NONE
+     *  |   +-- B.txt (file) NONE
+     */
+
+    const auto testNode =
+            std::make_shared<Node>(std::nullopt, _syncPal->updateTree(side)->side(), Str2SyncName("test"), NodeType::Directory,
```

Comment:
We use this constructor so many times with the first argument equal to `std::nullopt` that we could define a new constructor with a shorter list of arguments; at least for the unit tests.

Reply:
Done here: [71a448f](https://github.com/Infomaniak/desktop-kDrive/pull/347/commits/71a448f59ceecf81771afb75e80875e7df4af8cd)
Merge state
Review required
At least 1 approving review is required by reviewers with write access. 
[View](https://github.com/Infomaniak/desktop-kDrive/pull/347/files#conversations-menu)
Unresolved conversations
9 conversations must be resolved before merging.
Some checks haven’t completed yet
1 queued and 3 expected checks
@github-actions
Dependent Issues / check (pull_request_target) Queued — Waiting to run this check...
[Details](https://github.com/Infomaniak/desktop-kDrive/actions/runs/11341688325/job/31540470169?pr=347)
Dependent Issues Expected — Waiting for status to be reported
Required
SonarCloud Code Analysis Expected — Waiting for status to be reported
Required
build-kDrive Expected — Waiting for status to be reported
Required
This branch is out-of-date with the base branch
Merge the latest changes from develop into this branch.
 Merge without waiting for requirements to be met (bypass branch protections)

 Automatically merge when all requirements are met. [Learn more](https://docs.github.com/github/collaborating-with-issues-and-pull-requests/automatically-merging-a-pull-request)
@herve-er


Add a comment
Comment
 
Add your comment here...
 
Remember, contributions to this repository should follow our GitHub Community Guidelines.
 ProTip! Add .patch or .diff to the end of URLs for Git’s plaintext views.
Reviewers
@[ClementKunz](https://github.com/ClementKunz)
ClementKunz

@[luc-guyot-infomaniak](https://github.com/luc-guyot-infomaniak)
luc-guyot-infomaniak

At least 1 approving review is required to merge this pull request.

Still in progress?
Assignees
@[herve-er](https://github.com/herve-er)
herve-er

Labels
None yet
Projects
None yet
Milestone
No milestone
Development
Successfully merging this pull request may close these issues.

None yet


Notifications
Customize
You’re receiving notifications because you were assigned.
3 participants
@herve-er
@ClementKunz
@luc-guyot-infomaniak